rct229/rulesets/ashrae9012019/section19/section19rule19.py

```diff
@@ -0,0 +1,275 @@
+]
+
+
+COOLING_SYSTEM = schema_enums["CoolingSystemOptions"]
+REQ_FAN_POWER_FLOW_RATIO = 0.3 * ureg("W/cfm")
+
+
+class Section19Rule19(RuleDefinitionListIndexedBase):
+    """Rule 19 of ASHRAE 90.1-2019 Appendix G Section 19 (HVAC - General)"""
+
+    def __init__(self):
+        super(Section19Rule19, self).__init__(
+            rmrs_used=UserBaselineProposedVals(False, True, True),
+            each_rule=Section19Rule19.HVACRule(),
+            index_rmr="baseline",
+            id="19-19",
+            description="For baseline systems 9 and 10 the system fan electrical power (Pfan) for supply, return, exhaust, and relief shall be CFMs × 0.3, where, CFMs = the baseline system maximum design supply fan airflow rate, cfm. "
+            "If modeling a non-mechanical cooling fan is required by Section G3.1.2.8.2, there is a fan power allowance of Pfan = CFMnmc × 0.054, where, CFMnmc = the baseline non-mechanical cooling fan airflow, cfm for the non-mechanical cooling.",
+            ruleset_section_title="HVAC - General",
+            standard_section="Section G3.1.2.9",
+            is_primary_rule=True,
+            rmr_context="ruleset_model_descriptions/0",
+            list_path="$.buildings[*].building_segments[*].heating_ventilating_air_conditioning_systems[*]",
+        )
+
+    def is_applicable(self, context, data=None):
+        rmd_b = context.baseline
+        baseline_system_types_dict = get_baseline_system_types(rmd_b)
+
+        return any(
+            [
+                baseline_system_type_compare(system_type, applicable_sys_type, False)
+                for system_type in baseline_system_types_dict
+                for applicable_sys_type in APPLICABLE_SYS_TYPES
+            ]
+        )
+
+    def create_data(self, context, data):
+        rmd_b = context.baseline
+        rmd_p = context.proposed
+
+        baseline_system_types_dict = get_baseline_system_types(rmd_b)
+        applicable_hvac_sys_ids = [
+            hvac_id
+            for sys_type in baseline_system_types_dict
+            for target_sys_type in APPLICABLE_SYS_TYPES
+            if baseline_system_type_compare(sys_type, target_sys_type, False)
+            for hvac_id in baseline_system_types_dict[sys_type]
+        ]
+
+        dict_of_zones_and_terminal_units_served_by_hvac_sys_b = (
+            get_dict_of_zones_and_terminal_units_served_by_hvac_sys(rmd_b)
+        )
+
+        zone_hvac_has_non_mech_cooling_p = any(
+            [
+                find_one(
+                    "$.cooling_system.cooling_system",
+                    find_exactly_one_hvac_system(rmd_p, hvac_id_p),
+                )
+                == COOLING_SYSTEM.NON_MECHANICAL
+                for zone_id_p in find_all(
+                    "$.buildings[*].building_segments[*].zones[*].id", rmd_p
+                )
+                for hvac_id_p in get_list_hvac_systems_associated_with_zone(
+                    rmd_p, zone_id_p
+                )
+            ]
+        )
```

Comment:
Shouldn't this be tied to hvac systems because the RDS says ```Check if any of the hvac systems serving the zone have non-mechanical cooling as the cooling_system type```? 

If this is tied to each zone_id, I believe we also need to use the ```served_by_heating_ventilating_air_conditioning_system``` to know which hvac system serves each terminal. Then, we need to do the rule check at the terminal level, which I think is not the intention of this rule. What do you think?